usr/src/uts/common/fs/zfs/txg.c

```diff
@@ -446,12 +447,14 @@ txg_dispatch_callbacks(dsl_pool_t *dp, uint64_t txg)
 static void
 txg_sync_thread(dsl_pool_t *dp)
 {
+	char name[TXG_THREAD_MAXNAMELEN];
 	spa_t *spa = dp->dp_spa;
 	tx_state_t *tx = &dp->dp_tx;
 	callb_cpr_t cpr;
 	uint64_t start, delta;
 
-	txg_thread_enter(tx, &cpr);
+	snprintf(name, TXG_THREAD_MAXNAMELEN, "txg_sync %s", spa_name(spa));
```

Comment:
This will be much less of an issue when changed to CB_MAXNAME (31).  But if it's shorter on other platforms, I'm OK with changing it.  Maybe even <poolname>-txg_sync?